book.md

```diff
@@ -680,7 +679,255 @@ signal handler.  The ability to fake a signal handler frame and call
+be used to mitigate code reuse attacks. Pointer Authentication introduces
+instructions that generate a pointer _signature_, called a Pointer
+Authentication Code (PAC), based on a key and a modifier, and matching
+instructions to authenticate this signature. Incorrect authentication leads to
+an unusable pointer, that will cause a fault when used [^fpac]. The key is not
+directly accessible by user space software.
+
+[^fpac]: With the FPAC extension, a fault is raised at incorrect authentication.
+
+[Clang](https://clang.llvm.org/docs/ClangCommandLineReference.html#aarch64) and
+[GCC](https://gcc.gnu.org/onlinedocs/gcc/AArch64-Options.html) both use Pointer
+Authentication for return address signing, when compiling with the
+`-mbranch-protection=pac-ret` flag. When compiling with Clang using this flag,
+the `main` function from the [earlier stack buffer overflow
+example](#stack-buffer-overflow) looks like:
+
+```
+main:
+    cmp w0, #2
+    b.lt    .LBB1_2
+    paciasp
+    stp x29, x30, [sp, #-16]!
+    ldr x0, [x1, #8]
+    mov x29, sp
+    bl  copy_and_print
+    ldp x29, x30, [sp], #16
+    autiasp
+.LBB1_2:
+    mov w0, wzr
+    ret
+```
+
+Notice the `paciasp` and `autiasp` instructions: `paciasp` computes a PAC for
+the return address in the link register (`x30`), based on the current value of
+the stack pointer (`sp`) and a key. This PAC is inserted in the extension bits
+of the pointer. We then store this signed version of the link register on the
+stack.  Before returning, we load the signed return address from the stack, we
+execute `autiasp`, which verifies the PAC stored in the return address, again
+based on the value of the key and the value of the stack pointer (which at this
+point will be the same as when we signed the return address). If the PAC is
+correct, which will be the case in normal execution, the extension bits of the
+address are restored, so that the address can be used in the `ret` instruction.
+However, if the stored return address has been overwritten with an address with
+an incorrect PAC, the upper bits will be corrupted so that subsequent uses of
+the address (such as in the `ret` instruction) will result in a fault.  By
+making sure we don't store any return addresses without a PAC, we can
+significantly reduce the effectiveness of ROP attacks.
+
+Pointer Authentication can be used more widely, to also implement a
+forward-edge CFI scheme, as is done in the arm64e ABI [@McCall2019].
+
+### BTI
+
+[Branch Target Identification
+(BTI)](https://developer.arm.com/documentation/102433/0100/Jump-oriented-programming?lang=en)
+\index{BTI}, introduced in Armv8.5, offers coarse-grained forward-edge
+protection. With BTI, the locations that are targets of indirect branches have
+to be marked with a new instruction, `BTI`. There are a few different types of
+BTI instructions that permit different types of indirect branches (`BR`, `BLR`,
+or both).
```

Comment:
I'd say at then end "(indirect jump, indirect call, or both)". I would hope more readers would understand that as not all readers might be very familiar with AArch64 assembly.